scripts/hooks/tech-lead-authoring-guard.py

```
@@ -0,0 +1,479 @@
+            # Pick trailing tokens that look like paths.
+            for token in reversed(tokens):
+                if token.startswith("-") or token in {"sed", "awk", "gawk", "perl", "ruby", "inplace"}:
+                    continue
+                # Skip the -i argument's quoted backup-suffix value if any.
+                if "/" in token or "." in token or not token.startswith("'"):
+                    targets.append(token)
+                    break
+    return targets
+
+
+def _extract_mutation_targets(command: str):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `_extract_mutation_targets`

(ANN202)